src/Xamarin.CodeAnalysis/XI0004.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.
+//
+// RequiresSuperAttributeAnalyzer.cs:
+//
+// Authors:
+//   Madeline Zhang  <v-madzh@microsoft.com>
+//
+// Copyright 2020 Xamarin Inc.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.Formatting;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace RequiresSuperAttribute
+{
+	/// <summary>
+	/// Analyzer that reports diagnostics for all methods with the [RequiresSuper] attribute which are overridden without a base.[methodname]() call in the method body
+	/// </summary>
+	[DiagnosticAnalyzer (LanguageNames.CSharp)]
+	public class XI0004 : DiagnosticAnalyzer
```

Comment:
And let's add the `RequiresSuperAttribute` suffix to the analyzer ID, so it's named `XI0004RequiresSuperAttribute`, so it's easy to find in the future by just navigating code for either the analyzer code or the "description" of what it does.

Reply:
That should match the file name I suggest `XIA1004RequiresSuper` to follow the convention of other analyzers